Improve spacing for two argument predicates

dlesbre opened the pull request:
I found that `isList l xs` and `map f l` was a bit hard to read in the latex as it tended to squash the space between both arguments. Hopefully this version is more legible.

jihgfee commented:
The fact that we do not have an `\isList` or `map` is quite odd. Good find!

I think it would be better to place the commands in `macros.tex`.

Additionally, we mainly use `\spac` (defined in `iris.sty`) rather than `~` for uniform spacing.

So I suggest macros like the following:
`\newcommand{\isList}[2]{\operatorname{isList}#1\spac#2}`

Finally, it would be nice if you could guarantee that all occurrences of `IsList` and `map`.

There also seems to be other instances of this problem (e.g isStack), so it would be good to get an overview of those for starters, and then unify them later. We can make an issue for that when we close this MR.

Oh, and dont forget to add your name to the list of contributors in `main.tex` ;)

dlesbre commented:
Okay, I moved the macros to `macro.tex`, I also added a two others I found (`listFilter` and `all`) and some conversion I forgot. 

I'm pretty sure I converted all occurrences of these four, checked by running `grep -r "\\operatorname{isList}"`. It only prints the macro definition and places where the predicate is referenced without arguments.

As far as other such functions, `isStack` tends to use `\mathrm` instead of `\operatorname` and the parenthesis function notation `isStack(foo, bar)` instead of the ML `isStack foo bar` so it is fairly easy to read without spaces (but also, it is ~~infuriating~~*mildly annoying* to have a different notation).

For others I also spotted an quantified `I` variable in [case-study-folder.tex](./sections/case-study-folder.tex) which has the same two argument syntax. It renders fine since spaces are manually inserted everywhere it is called.

jihgfee commented:
LGTM! Thanks for the MR :)!

Good find about the discrepancy on `isStack` - I'll make an issue about it, and then we can fix it separately.
